smarts/env/gymnasium/utils/observation_conversion.py

```diff
@@ -0,0 +1,1050 @@
+
+    @property
+    def name(self):
+        return "drivable_area_grid_map"
+
+
+class LidarPointCloudSpaceFormat(ConfigurableSpaceFormat):
+    def format(self, obs: Observation):
+        lidar_point_cloud = obs.lidar_point_cloud
+        # # MTA TODO: add lidar configuration like following:
+        # sensor_params = self._agent_interface.lidar_point_cloud.sensor_params
+        # n_rays = int(
+        #     (sensor_params.end_angle - sensor_params.start_angle)
+        #     / sensor_params.angle_resolution
+        # )
+        des_shp = _LIDAR_SHP
+        hit = np.array(lidar_point_cloud[1], dtype=np.int8)
+        point_cloud = np.array(lidar_point_cloud[0], dtype=np.float64)
+        point_cloud = np.nan_to_num(
+            point_cloud,
+            copy=False,
+            nan=np.float64(0),
+            posinf=np.float64(0),
+            neginf=np.float64(0),
+        )
+        ray_origin, ray_vector = zip(*(lidar_point_cloud[2]))
+        ray_origin = np.array(ray_origin, np.float64)
+        ray_vector = np.array(ray_vector, np.float64)
+
+        try:
+            assert hit.shape == (des_shp,)
+            assert point_cloud.shape == (des_shp, 3)
+            assert ray_origin.shape == (des_shp, 3)
+            assert ray_vector.shape == (des_shp, 3)
+        except Exception as exc:
+            raise Exception(
+                "Internal Error: Mismatched lidar point cloud shape."
+            ) from exc
+
+        return {
+            "hit": hit,
+            "point_cloud": point_cloud,
+            "ray_origin": ray_origin,
+            "ray_vector": ray_vector,
+        }
+
+    def active(self, agent_interface: AgentInterface) -> bool:
+        return bool(agent_interface.lidar_point_cloud)
+
+    @property
+    def name(self):
+        return "lidar_point_cloud"
+
+    @property
+    def space(self):
+        # MTA TODO: add lidar configuration
+        return gym.spaces.Dict(
+            {
+                "hit": gym.spaces.MultiBinary(_LIDAR_SHP),
+                "point_cloud": gym.spaces.Box(
+                    low=-1e10, high=1e10, shape=(_LIDAR_SHP, 3), dtype=np.float64
+                ),
+                "ray_origin": gym.spaces.Box(
+                    low=-1e10, high=1e10, shape=(_LIDAR_SHP, 3), dtype=np.float64
+                ),
+                "ray_vector": gym.spaces.Box(
+                    low=-1e10, high=1e10, shape=(_LIDAR_SHP, 3), dtype=np.float64
+                ),
+            }
+        )
+
+
+class MissionSpaceFormat(BaseSpaceFormat):
+    def format(self, obs: Observation):
+        ego_vehicle_obs = obs.ego_vehicle_state
+        if hasattr(ego_vehicle_obs.mission.goal, "position"):
+            goal_pos = np.array(ego_vehicle_obs.mission.goal.position, dtype=np.float64)
+        else:
+            goal_pos = np.zeros((3,), dtype=np.float64)
+
+        return {"goal_position": goal_pos}
+
+    def active(self, agent_interface: AgentInterface) -> bool:
+        return True
```

Comment:
We have `EndlessGoal`, `PositionalGoal`, and `TraverseGoal`, but only the `PositionalGoal` is visible (through its desired final position) in the returned observation. Should boolean return value of `active()` be dependent on whether the goal is a `PositionalGoal`?

Reply:
Goal is determined from the scenario, so it is not possible to determine it from the interface.